prow/moonraker/client.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package moonraker
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+)
+
+type Client struct {
+	host       string
+	httpClient *http.Client
+}
+
+func NewClient(host string, timeout time.Duration) *Client {
+	return &Client{
+		host: host,
+		httpClient: &http.Client{
+			Timeout: timeout,
+		},
+	}
+}
+
+func (c *Client) do(method, endpoint string, payload []byte, params map[string]string) (*http.Response, error) {
+	baseURL := fmt.Sprintf("%s/%s", c.host, endpoint)
+	req, err := http.NewRequest(method, baseURL, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json; charset=UTF-8")
+	q := req.URL.Query()
+	for key, val := range params {
+		q.Set(key, val)
+	}
+	req.URL.RawQuery = q.Encode()
+	return c.httpClient.Do(req)
+}
+
+// GetProwYAML returns the inrepoconfig contents for a repo, based on the Refs
+// struct as the input. From the Refs, Moonraker can determine the org/repo,
+// BaseSHA, and the Pulls[] (additional refs of each PR, if any) to grab the
+// inrepoconfig contents.
+func (c *Client) GetProwYAML(refs *prowapi.Refs) (*config.ProwYAML, error) {
+	buf, err := json.Marshal(refs)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal %v", refs)
+	}
+
+	resp, err := c.do(http.MethodPost, "get-inrepoconfig", buf, nil)
```

Comment:
We should probably make the `get-inrepoconfig` path component a constant to avoid typos or skew.